community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -1085,6 +1085,7 @@ public synchronized void stop()
         // Here we're zooming in and focusing on getting committed transactions to close.
         awaitAllTransactionsClosed();
         LogFile logFile = transactionLogModule.logFile();
+        life.stop();
```

Comment:
This feels a bit hacky in that we stop everything that is in `life`, including `PhysicalLogFile` and `TransactionAppender` and all such services, to then go ahead and write a checkpoint, using those services. It works due to trickery with stop vs shutdown for PhysicalLogFile, and luckily TransactionAppender has neither stop nor shutdown. It may make future changes harder to understand. Can we instead get by just stopping the checkpointer, or perhaps other select services that we know would interfere with what we are about to do below?